services/galley/src/Galley/API/Action.hs

```diff
@@ -359,23 +359,24 @@ instance IsConversationAction ConversationAccessData where
         r
   conversationAction = ConversationActionAccessUpdate
   ensureAllowed _ target conv self = do
-    -- 'PrivateAccessRole' is for self-conversations, 1:1 conversations and
-    -- so on; users are not supposed to be able to make other conversations
-    -- have 'PrivateAccessRole'
-    when
-      ( PrivateAccess `elem` cupAccess target
-          || PrivateAccessRole == cupAccessRole target
-      )
-      $ throw InvalidTargetAccess
-    -- Team conversations incur another round of checks
-    case convTeam conv of
-      Just _ -> do
-        -- Access mode change might result in members being removed from the
-        -- conversation, so the user must have the necessary permission flag
-        ensureActionAllowed RemoveConversationMember self
-      Nothing ->
-        when (cupAccessRole target == TeamAccessRole) $
-          throw InvalidTargetAccess
+    -- -- 'PrivateAccessRole' is for self-conversations, 1:1 conversations and
+    -- -- so on; users are not supposed to be able to make other conversations
+    -- -- have 'PrivateAccessRole'
+    -- when
+    --   ( PrivateAccess `elem` cupAccess target
+    --       || PrivateAccessRole == cupAccessRole target
+    --   )
+    --   $ throw InvalidTargetAccess
+    -- -- Team conversations incur another round of checks
+    -- case convTeam conv of
+    --   Just _ -> do
+    --     -- Access mode change might result in members being removed from the
+    --     -- conversation, so the user must have the necessary permission flag
+    --     ensureActionAllowed RemoveConversationMember self
+    --   Nothing ->
+    --     when (cupAccessRole target == TeamAccessRole) $
+    --       throw InvalidTargetAccess
+    error "todo(leif):implement"
```

Comment:
If at all possible could you try to change the indentation less?  Makes it much easier to read.